test/integration/targets/hwc_network_vpc/tasks/main.yml

```
@@ -0,0 +1,98 @@
+---
+# ----------------------------------------------------------------------------
+#
+#     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+#
+# ----------------------------------------------------------------------------
+#
+#     This file is automatically generated by Magic Modules and manual
+#     changes will be clobbered when the file is regenerated.
+#
+#     Please read more about how to change this file at
+#     https://www.github.com/huaweicloud/magic-modules
+#
+# ----------------------------------------------------------------------------
+# Pre-test setup
+- name: delete a vpc
+  hwc_network_vpc:
+      identity_endpoint: "{{ identity_endpoint }}"
+      user_name: "{{ user_name }}"
+      password: "{{ password }}"
+      domain_name: "{{ domain_name }}"
+      project_name: "{{ project_name }}"
+      region: "{{ region }}"
+      name: "vpc_1"
+      cidr: "192.168.100.0/24"
+      state: absent
+#----------------------------------------------------------
+- name: create a vpc
+  hwc_network_vpc:
+      identity_endpoint: "{{ identity_endpoint }}"
+      user_name: "{{ user_name }}"
+      password: "{{ password }}"
+      domain_name: "{{ domain_name }}"
+      project_name: "{{ project_name }}"
+      region: "{{ region }}"
+      name: "vpc_1"
+      cidr: "192.168.100.0/24"
+      state: present
+  register: result
+- name: assert changed is true
+  assert:
+    that:
+      - result.failed == 0
+      - result.changed == true
```

Instead of these two lines, you can simply write
```suggestion
      - result is changed
```
That is only true if the module has not failed (that can be abbreviated by `result is not failed`) and has `changed` set.

Also, when doing tests, it makes sense to also test idempotency, i.e. execute the module with the same options again, and check that it didn't change something. If your module supports check mode, you can do four invocations:
1. call module with check mode (this should be changed);
2. call module without check mode (this should be changed);
3. call module with check mode (this should not be changed);
4. call module without check mode (this should not be changed).

That will make sure that check mode behaves the same as without check mode.